scaleway/resource_mnq_sqs_test.go

```diff
@@ -0,0 +1,178 @@
+package scaleway
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	accountV3 "github.com/scaleway/scaleway-sdk-go/api/account/v3"
+	mnq "github.com/scaleway/scaleway-sdk-go/api/mnq/v1beta1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func init() {
+	resource.AddTestSweepers("scaleway_mnq_sqs", &resource.Sweeper{
+		Name: "scaleway_mnq_sqs",
+		F:    testSweepMNQSQS,
+	})
+}
+
+func testSweepMNQSQS(_ string) error {
+	return sweepRegions((&mnq.SqsAPI{}).Regions(), func(scwClient *scw.Client, region scw.Region) error {
+		accountAPI := accountV3.NewProjectAPI(scwClient)
+		mnqAPI := mnq.NewSqsAPI(scwClient)
+
+		l.Debugf("sweeper: destroying the mnq sqss in (%s)", region)
+
+		listProjects, err := accountAPI.ListProjects(&accountV3.ProjectAPIListProjectsRequest{}, scw.WithAllPages())
+		if err != nil {
+			return fmt.Errorf("failed to list projects: %w", err)
+		}
+		for _, project := range listProjects.Projects {
+			if !strings.HasPrefix(project.Name, "tf_tests") {
+				continue
+			}
+
+			_, err := mnqAPI.DeactivateSqs(&mnq.SqsAPIDeactivateSqsRequest{
+				Region:    region,
+				ProjectID: project.ID,
+			})
+			if err != nil {
+				l.Debugf("sweeper: error (%s)", err)
+				return err
+			}
+		}
+
+		return nil
+	})
+}
+
+func TestAccScalewayMNQSQS_Basic(t *testing.T) {
+	tt := NewTestTools(t)
+	defer tt.Cleanup()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy:      testAccCheckScalewayMNQSQSDestroy(tt),
+		Steps: []resource.TestStep{
+			{
+				Config: `
+					resource scaleway_account_project main {}
+
+					resource scaleway_mnq_sqs main {
+						project_id = scaleway_account_project.main.id
+					}
+				`,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayMNQSQSExists(tt, "scaleway_mnq_sqs.main"),
+					testCheckResourceAttrUUID("scaleway_mnq_sqs.main", "id"),
+					resource.TestCheckResourceAttrSet("scaleway_mnq_sqs.main", "endpoint"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccScalewayMNQSQS_AlreadyActivated(t *testing.T) {
```

Comment:
Could you add a test that test whether I can activate SQS and Nats in a different region than the one specified in the provider?

Reply:
Only fr-par is currently available